stock_sale_product_volume/models/stock_package_type.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2024 Akretion (<http://www.akretion.com>).
+# @author Mathieu Delva <mathieu.delva@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPackageType(models.Model):
+    _inherit = "stock.package.type"
+
+    volume = fields.Float(
+        compute="_compute_volume",
+        store=True,
+    )
+    volume_uom_name = fields.Char(
+        string="Volume unit of measure label", compute="_compute_volume_uom_name"
+    )
+
+    def _compute_length_uom_name(self):
+        self.length_uom_name = self.env[
+            "product.template"
+        ]._get_length_uom_name_from_ir_config_parameter()
+
+    def _compute_volume_uom_name(self):
+        self.volume_uom_name = self.env[
+            "product.template"
+        ]._get_volume_uom_name_from_ir_config_parameter()
+
+    @api.depends("packaging_length", "width", "height")
+    def _compute_volume(self):
+        conversion_uom_dict = {"mm³": 1, "cm³": 1000000, "m³": 1000000000}
```

Comment:
convention nitpicking: you can leave constants at the top level of the file in capital letters. Otherwise LGTM